src/components/pages/StudentDashboard/StudentViewRumble/StudentRumbleRedirect.tsx

```diff
@@ -20,54 +21,35 @@ const StudentRumbleRedirect = ({
   const [submission, setSubmission] = useRecoilState(current.sub);
   // Whether or not the user has given feedback to others yet
   const [feedbackComplete, setFeedbackComplete] = useState<boolean>();
-  const [loading, setLoading] = useState(true);
+  // const [loading, setLoading] = useState(true);
 
   const hasSubmitted = useRecoilValue(current.hasSubmitted);
 
+  const [getSubForRumble, loading, ,] = useAsync({
+    asyncFunction: Students.getSubForRumble,
+    setter: setSubmission,
+  });
+
   // This useEffect is loading the current user's submission for the rumble
   useEffect(() => {
     if (rumble && user && !submission) {
-      setLoading(true);
-      Students.getSubForRumble(rumble.id, user.id)
-        .then((res) => {
-          console.log('subs for rumble', { res });
-          // Set loading to false, submission will also be false, so the
-          // student submission page will be rendered
-          if (res === undefined) setLoading(false);
-          // If submission is defined after this, it will run the next useEffect
-          setSubmission(res);
-        })
-        .catch((err) => {
-          console.log({ err });
-          // setError('There is no submission for this Rumble.');
-        })
-        .finally(() => {
-          setLoading(false);
-        });
+      getSubForRumble(rumble.id, user.id);
```

Comment:
This page shouldn't use the useAsync hook because we're tracking loading states in a non-default way!